contrib/init/upstart/docker.conf

```diff
@@ -19,17 +23,15 @@ pre-start script
 	if ! mountpoint -q /sys/fs/cgroup; then
 		mount -t tmpfs -o uid=0,gid=0,mode=0755 cgroup /sys/fs/cgroup
 	fi
-	(
-		cd /sys/fs/cgroup
-		for sys in $(awk '!/^#/ { if ($4 == 1) print $1 }' /proc/cgroups); do
-			mkdir -p $sys
-			if ! mountpoint -q $sys; then
-				if ! mount -n -t cgroup -o $sys cgroup $sys; then
-					rmdir $sys || true
-				fi
+
+	for sys in $(awk '!/^#/ { if ($4 == 1) print $1 }' /proc/cgroups); do
+		mkdir -p /sys/fs/cgroup/$sys
+		if ! mountpoint -q /sys/fs/cgroup/$sys; then
```

Comment:
if `mountpoint` is going to be a shell function we provide, there's no need for the `-q`.